src/main/flight/pid.c

```diff
@@ -287,23 +293,76 @@ void pidResetIterm(void)
     }
 }
 
+
 #ifdef USE_WING
-static float getWingTpaArgument(float throttle)
+static float calcWingThrottle(void)
+{
+    float batteryThrottleFactor = 1.0f;
+    if (pidRuntime.tpaSpeed.maxVoltage > 0.0f) {
+        batteryThrottleFactor = getBatteryVoltageLatest() / 100.0f / pidRuntime.tpaSpeed.maxVoltage;
+        batteryThrottleFactor = constrainf(batteryThrottleFactor, 0.0f, 1.0f);
+    }
+
+    return getAverageMotorOutput() * batteryThrottleFactor;
+}
+
+
+static float calcWingAccelerationAdvanced(float throttle, float pitchAngle)
+{
+    const tpaSpeedParams_t *tpa = &pidRuntime.tpaSpeed;
+
+    const float thrust = (throttle * throttle - throttle * tpa->speed / tpa->propMaxSpeed) * tpa->twr * G_ACCELERATION;
```

Comment:
```suggestion
    const float thrust = (throttle - tpa->speed / tpa->propMaxSpeed) * tpa->twr * G_ACCELERATION;
```

Reply:
@KarateBrot please check the math here :)

Reply:
t * t - t * A is not equal to t - A :)

Reply:
Yeah you are correct. Only works when t == 1.0f :rofl:

Reply:
Mark is only going full throttle, i like it :)